druid/src/widget/split.rs

```diff
@@ -88,38 +98,89 @@ impl<T> Split<T> {
     pub fn fill_splitter_handle(mut self, solid: bool) -> Self {
         self.solid = solid;
         self
     }
-    fn splitter_hit_test(&self, size: Size, mouse_pos: Point) -> bool {
+
+    /// Returns the width of the area of mouse hit detection.
+    #[inline]
+    fn splitter_area(&self) -> f64 {
+        self.splitter_size.max(self.min_splitter_area)
+    }
+
+    /// Returns the padding width added to the splitter on both sides.
+    #[inline]
+    fn splitter_padding(&self) -> f64 {
+        (self.splitter_area() - self.splitter_size) / 2.0
+    }
+
+    /// Returns the location of the edges of the splitter area,
+    /// given the specified total size.
+    fn splitter_edges(&self, size: Size) -> (f64, f64) {
+        let splitter_area = self.splitter_area();
         match self.split_direction {
             Axis::Vertical => {
-                let center = size.width * self.split_point;
-                (center - mouse_pos.x).abs() < self.splitter_size.min(5.0) / 2.0
+                let reduced_width = size.width - splitter_area;
+                let edge1 = (reduced_width * self.split_point_effective).floor();
+                let edge2 = edge1 + splitter_area;
+                (edge1, edge2)
             }
             Axis::Horizontal => {
-                let center = size.height * self.split_point;
-                (center - mouse_pos.y).abs() < self.splitter_size.min(5.0) / 2.0
+                let reduced_height = size.height - splitter_area;
+                let edge1 = (reduced_height * self.split_point_effective).floor();
+                let edge2 = edge1 + splitter_area;
+                (edge1, edge2)
             }
         }
     }
+
+    /// Returns true if the provided mouse pos is inside the splitter area.
+    fn splitter_hit_test(&self, size: Size, mouse_pos: Point) -> bool {
+        let (edge1, edge2) = self.splitter_edges(size);
+        match self.split_direction {
+            Axis::Vertical => mouse_pos.x >= edge1 && mouse_pos.x <= edge2,
+            Axis::Horizontal => mouse_pos.y >= edge1 && mouse_pos.y <= edge2,
+        }
+    }
+
+    /// Returns the min and max split coordinate of the provided size.
     fn calculate_limits(&self, size: Size) -> (f64, f64) {
         // Since the Axis::Direction tells us the direction of the splitter itself
-        // we need the minor axis to get the size of the 'splitted' direction
-        let size_in_splitted_direction = self.split_direction.minor(size);
+        // we need the minor axis to get the size of the split direction
+        let size_in_split_direction = self.split_direction.minor(size);
 
-        let min_offset = (self.splitter_size * 0.5).min(5.0);
-        let mut min_limit = self.min_size.max(min_offset);
-        let mut max_limit = (size_in_splitted_direction - self.min_size.max(min_offset)).max(0.0);
+        let mut min_limit = self.min_size;
+        let mut max_limit = (size_in_split_direction - min_limit).max(0.0);
```

Comment:
I see we've removed the absolute '5.0' minimum. This was originally added to 'fix' the possibility that the splitter would get 'lost' in the edge of it's container - or that nested splitters could collide.
I'm okay with removing it if the behavior is still acceptable in these extrema, but it's something worth checking for.

Reply:
This minimum isn't needed with the new logic, because the splitter bar is guaranteed to always be visible. This specific function calculates the min/max limits given the provided size and doesn't need to know about the splitter bar at all. The bar logic is elsewhere.